src/rgw/rgw_auth_s3.cc

```diff
@@ -446,7 +452,11 @@ static inline int parse_v4_auth_header(const req_info& info,               /* in
   }
 
   if (info.env->exists("HTTP_X_AMZ_SECURITY_TOKEN")) {
-    sessiontoken = info.env->get("HTTP_X_AMZ_SECURITY_TOKEN");
+    boost::optional<const std::string&> token;
```

Comment:
if you use `get_optional()` you probably don't need to also do the check in line 454.
this would also save on doing the lookup twice